ntoskrnl/io/iomgr/deviface.c

```diff
@@ -257,16 +523,12 @@ IoOpenDeviceInterfaceRegistryKey(IN PUNICODE_STRING SymbolicLinkName,
 
     InitializeObjectAttributes(&ObjectAttributes,
                                &DeviceParametersU,
-                               OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE | OBJ_OPENIF,
+                               OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE,
                                InstanceKey,
                                NULL);
-    Status = ZwCreateKey(&DeviceParametersKey,
-                         DesiredAccess,
-                         &ObjectAttributes,
-                         0,
-                         NULL,
-                         REG_OPTION_NON_VOLATILE,
-                         NULL);
+    Status = ZwOpenKey(&DeviceParametersKey,
```

Comment:
That were a changes from my previous unsuccessful experiments. I though that the keys shouldn't be created here if they don't exist, but I'm not sure about this exactly. My kmtest allows to check whether the key is volatile or not, but I'm not sure that I can check is it created or not in case it does not exist, because Windows has more device classes registered and there seems no any missing of them. Maybe I can try to pass non-existing class into here in my kmtest, and see is the key created then.
In any case, you're right: since I'm not fixing this function in this PR, it indeed should be sent in a separate PR, even if it's required. Then I'll revert it here.

Reply:
Done. :ok_hand: